src/System.Private.ServiceModel/tools/SelfHostedWcfService/dir.targets

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" InitialTargets="_NugetPackageRestore" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Inline task to bootstrap the build to enable downloading nuget.exe -->
+  <UsingTask TaskName="DownloadFile" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <Address ParameterType="System.String" Required="true" />
+      <FileName ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System" />
+      <Reference Include="System.IO" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+            var directory = System.IO.Path.GetDirectoryName(FileName);
+            Directory.CreateDirectory(directory);
+
+            var tempFile = Path.Combine(directory, Path.GetRandomFileName());
+            var client = new System.Net.WebClient();
+            client.Proxy = System.Net.WebRequest.DefaultWebProxy;
+            if (client.Proxy != null) client.Proxy.Credentials = System.Net.CredentialCache.DefaultCredentials;
+            var tryCount = 1;
+            var maxTries = 3;
+
+            while (tryCount <= maxTries)
+            {
+                try
+                {
+                    Log.LogMessage("Attempting to download {0}...", Address);
+                    client.DownloadFile(Address, tempFile);
+                    break;
+                }
+                catch (System.Net.WebException e)
+                {
+                    tryCount++;
+                    if (tryCount > maxTries)
+                    {
+                        throw;
+                    }
+                    else
+                    {
+                        Log.LogMessage(MessageImportance.High, "Download failed, retrying: {0}", e.Message);
+                    }
+                }
+            }
+
+            try
+            {
+                if (!File.Exists(FileName))
+                    File.Move(tempFile, FileName);
+            }
+            finally
+            {
+                if (File.Exists(tempFile))
+                    File.Delete(tempFile);
+            }
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <Target Name="_NugetPackageRestore" DependsOnTargets="_DownloadAndRunNugetPackageRestore" />
+
+  <Target Name="_DownloadAndRunNugetPackageRestore">
+    <Message Importance="High" Text="Restoring nuget packages..." />
+
+    <!-- Download latest nuget.exe -->
+    <DownloadFile FileName="$(NuGetToolPath)"
+                  Address="https://www.nuget.org/nuget.exe" />
+
+    <PropertyGroup>
+      <NugetRestoreCommand>"$(NuGetToolPath)" restore -PackagesDirectory $(PackageOutputDir)</NugetRestoreCommand>
+    </PropertyGroup>
+    
+    <Exec Command="$(NugetRestoreCommand)" StandardOutputImportance="Low" WorkingDirectory="$(SolutionDir)" />
+  </Target>
+</Project>
```

Comment:
By any chance, is there a higher-level dir.targets we can reference to get the nuget restore functionality?